rest/persistent_config_test.go

```diff
@@ -1493,15 +1493,18 @@ func persistentConfigTestCases() []persistentConfigTestCase {
 				xattrConfig: false,
 			},
 		}
-	} else {
-		return []persistentConfigTestCase{
-			{
-				name:        "xattr_persistence",
-				xattrConfig: true,
-			}, {
-				name:        "document_persistence",
-				xattrConfig: false,
-			},
-		}
 	}
+	testCases := []persistentConfigTestCase{
+		{
+			name:        "document_persistence",
+			xattrConfig: false,
+		},
+	}
+	if base.TestUseXattrs() {
+		testCases = append(testCases, persistentConfigTestCase{
+			name:        "xattr_persistence",
+			xattrConfig: true,
+		})
+	}
```

Comment:
This doesn't seem right and I'm curious if anything broke with this because there's an underlying bug, or whether this was just cleanup?

We should continue testing both unconditionally, because it's valid to set `UseXattrConfig` for SG config even when `"enable_shared_bucket_access": false` (which is what `SG_TEST_USE_XATTRS` is intended for)

Reply:
I thought this would just make it more obvious what was going on, testing xattr persistence is for capella and non xattr is not that common. This issue is fixed by switching the bucket emptier to `purgeErr := dataStore.DeleteWithXattrs(ctx, key, []string{base.SyncXattrName})` since what happens with xattr persistence tests is that the sync metadata is preserved between tests, and an earlier test is not cleaned up for a later test.

This is actually a bug with xattrs=false,views=true since the views bucket pool uses bucket flush and that's less of a problem.

Reply:
Readded these tests.